### DISCUSSION TITLE
How to get a tree of elements from IFC-file

### DISCUSSION THREAD

**Discussion by Babaiqa** [IfcOpenShell/IfcOpenShell#4825] [General]
Hi guys!))

Please help me figure it out. I myself specialize in Kotlin, and this is my first time encountering Python, so my problem may seem trivial. Despite this, I just can’t figure it out.

I need, having an IFC-file, to get a tree of all its elements from it, and present the final result in the form of a json. Something like that:
`{
  "id": "someGlobalId",
  "classIfc": "IfcBuildingStorey",
  "children": [
    {
      "id": "someGlobalId",
      "classIfc": "IfcWall"
    }
  ]
}`

Tried using the code from here
https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/src/ifcopenshell-python/ifcopenshell/geom/app.py#L263-L276

But I encounter an error that a required module is missing:
- ModuleNotFoundError: No module named 'OCC'
Somewhere I've read that it's pythonocc-core library and that it is not possible to pull up this library via pip, it is necessary
Conda, but I can’t use it to tighten up the current ifcopenshell.

Is there another solution? Or maybe I'm doing something wrong? I really don’t want to manually get all the elements, look for connections among them and try to make a tree myself.

I would really appreciate some help!))

**Comment by aothms** [comment]
You can use the geom/app.py module just as an example to traverse this tree, that part of the code doesn't actually depend on OCC, just don't import the module but copy paste the bits you need. Alternatively, a more modern way would be to use https://github.com/IfcOpenShell/IfcOpenShell/blob/03b2afbb31530518220cca9f6935b8766eaaea2d/src/ifcopenshell-python/ifcopenshell/util/element.py#L934
